doc/adrs/142_add_support_for_emf_models_migration.adoc

```diff
@@ -0,0 +1,104 @@
+    public MigrationExtendedMetaData(List<IMigrationParticipant> iMigrationParticipants, String version) {
+        this.iMigrationParticipants = Objects.requireNonNull(iMigrationParticipants);
+        this.version = version;
+    }
+
+    @Override
+    public EStructuralFeature getLocalElement(EClass eClass,  String namespace, String oldName) {
+        EStructuralFeature structuralFeature = eClass.getEStructuralFeature(oldName);
+        var migrationParticipants = iMigrationParticipants.stream().
+                filter(iMigrationParticipant -> iMigrationParticipant.canHandle(this.version))
+                .sorted(Comparator.comparing(IMigrationParticipant::getMigrationParticipantVersion))
+                .toList();
+
+        for (IMigrationParticipant migrationParticipant : migrationParticipants) {
+            structuralFeature = migrationParticipant.getLocalElement(eClass, namespace, oldName);
+        }
+
+        return structuralFeature;
+    }
+}
+----
+
+In each method overridden, we call the corresponding method from `IMigrationParticipant` if it has a lesser version than the one used to instantiate the `MigrationExtendedMetaData`.
+
+[source,java]
+----
+public class MigrationExtendedMetaData extends BasicExtendedMetaData {
+    boolean canHandle(String currentFileVersion);
+    String getMigrationParticipantVersion();
+    EStructuralFeature getLocalElement(EClass eClass,  String namespace, String oldName);
+}
+----
+
+[source,java]
+----
+public class MigrationExtendedMetaData extends BasicExtendedMetaData {
+    boolean canHandle(String currentFileVersion);
+    String getMigrationParticipantVersion();
+    EStructuralFeature getLocalElement(EClass eClass,  String namespace, String oldName);
+}
+----
+
+This is an example of implementation where this NameFeatureChangelMigrationParticipant will be used if the model version is superior to `202404221458`.
+If the feature name was changed from "name2" to something else, we provide the new correct value.
+
+[source,java]
+----
+@Service
+public class NameFeatureChangelMigrationParticipant implements IMigrationParticipant {
+
+    private static final String PARTICIPANT_VERSION = "202404221458";
+    @Override
+    public boolean canHandle(String currentFileVersion) {
+        return PARTICIPANT_VERSION.compareTo(currentFileVersion) < 0;
+    }
+
+    @Override
+    public String getMigrationParticipantVersion() {
+        return PARTICIPANT_VERSION;
+    }
+
+    @Override
+    public EStructuralFeature getAttribute(String namespace, String name) {
+        return null;
+    }
+
+    @Override
+    public EStructuralFeature getLocalElement(EClass eClass,  String namespace, String oldName) {
+        EStructuralFeature feature = null;
+        if (DomainPackage.eINSTANCE.getDomain().equals(eClass)) {
+            if (oldName.equals("name2")) {
+                feature = DomainPackage.eINSTANCE.getNamedElement_Name();
+            }
+        }
```

Comment:
Start thinking about the first concrete use case that this API should fulfill.

An attribute has disappeared (Node#labelExpression) and for those who had some expressions in this attribute, we need instead to create a new child (Node#insideLabel: InsideLabel) and set the expression on the newly created child (InsideLabel#labelExpression). I have a couple of other use cases in mind (there are a couple of attributes to rename for example but that's the first use case for the iteration).

No need to update the ADR for that use case, but when you will start implementing it, this is the first use case.